ethcore/wasm/src/lib.rs

```diff
@@ -121,15 +125,8 @@ impl vm::Vm for WasmInterpreter {
 			})?
 		);
 
-		let d_ptr = runtime.write_descriptor(
-			call_args::CallArgs::new(
-				params.address,
-				params.sender,
-				params.origin,
-				params.value.value(),
-				params.data.unwrap_or(Vec::with_capacity(0)),
-			)
-		).map_err(|e| Error(e))?;
+		let d_ptr = runtime.write_descriptor(params.data.unwrap_or(Vec::with_capacity(0)))
```

Comment:
`Vec::new` is guaranteed not to allocate. `unwrap_or_else` would be nicer looking for the same behavior